src/SSW.Rewards.Application/Achievements/Command/ClaimSocialMediaAchievementForUser/ClaimSocialMediaAchievementForUser.cs

```diff
@@ -23,24 +23,27 @@ public async Task<int> Handle(ClaimSocialMediaAchievementForUser request, Cancel
     {
         int? achievementId = await _context.SocialMediaPlatforms
                                           .AsNoTracking()
-                                          .Where(x => x.Id == request.SocialMediaPlatformId)
+                                          .Where(x => x.AchievementId == request.AchievementId)
                                           .Select(x => x.AchievementId)
                                           .FirstOrDefaultAsync(cancellationToken);
         if (achievementId == null)
             return 0;
         int currentUserId = await _userService.GetUserId(_currentUserService.GetUserEmail());
         var userAchievement = await _context.UserAchievements
-                                            .AsNoTracking()
                                             .Where(x => x.UserId == currentUserId && x.AchievementId == achievementId.Value)
                                             .FirstOrDefaultAsync(cancellationToken);
+
         if (userAchievement == null)
         {
+            var user = await _context.Users.FindAsync(currentUserId);
+            var achievement = await _context.Achievements.FindAsync(achievementId);
+
             userAchievement = new UserAchievement
             {
-                AchievementId = achievementId.Value,
-                UserId        = currentUserId,
-                AwardedAt     = _dateTimeService.Now,
-                CreatedUtc    = _dateTimeService.Now
+                Achievement = achievement,
+                User        = user, 
+                AwardedAt   = _dateTimeService.Now,
+                CreatedUtc  = _dateTimeService.Now
```

Comment:
You don't need to get this user object, or the achievement object. 2 unnecessary db round-trips.

Just set the userId and achievementId you already have.

Reply:
That's what I was originally doing and it caused an unexpected issue. It wouldn't record this entity in the db, and added a new user to the users table with no details. I changed it to work like this and it solved the problem.
I will add a comment saying why this is done this wat and a PBI to investigate.

Reply:
Done